File: src/main/java/com/paragon/client/ui/taskbar/Taskbar.java
package com.paragon.client.ui.taskbar;

import com.paragon.api.util.Wrapper;
import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.util.render.TextRenderer;
import com.paragon.client.systems.module.impl.client.ClientFont;
import com.paragon.client.ui.animation.Animation;
import com.paragon.client.ui.console.ConsoleGUI;
import com.paragon.client.ui.taskbar.icons.Icon;
import com.paragon.client.systems.module.impl.client.Colours;
import com.paragon.client.systems.module.impl.client.ClickGUI;
import net.minecraft.client.gui.ScaledResolution;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import java.awt.Color;
import java.util.ArrayList;
import java.util.Objects;

/**
 * @author Wolfsurge
 */
@SideOnly(Side.CLIENT)
public final class Taskbar implements Wrapper, TextRenderer {

    private ArrayList<Icon> icons = new ArrayList<>();

    private final Animation tooltipAnimation = new Animation(() -> 200f, false, ClickGUI.getEasing()::getValue);
    private String tooltip = "";

    public Taskbar() {
        int x = (int) getStringWidth("Paragon") + 10;

        icons.add(new Icon("GUI", x, ClickGUI::getGUI));
        x += getStringWidth("GUI") + 7;
        icons.add(new Icon("Console", x, ConsoleGUI::new));
    }

    public void drawTaskbar(int mouseX, int mouseY) {
        ScaledResolution scaledResolution = new ScaledResolution(mc);

        RenderUtil.drawRect(0, scaledResolution.getScaledHeight() - 18, scaledResolution.getScaledWidth(), 18, new Color(20, 20, 20).getRGB());
        RenderUtil.drawRect(0, scaledResolution.getScaledHeight() - 19, scaledResolution.getScaledWidth(), 2, Colours.mainColour.getValue().getRGB());

        renderText("Paragon", 2, scaledResolution.getScaledHeight() - (ClientFont.INSTANCE.isEnabled() ? 12 : 11), Colours.mainColour.getValue().getRGB());

        for (Icon icon : icons) {
            icon.draw(mouseX, mouseY);
        }

        if (tooltip != "" && ClickGUI.getTooltips().getValue()) {
            renderText(tooltip, (float) (scaledResolution.getScaledWidth() - ((getStringWidth(tooltip) + 2) * tooltipAnimation.getAnimationFactor())), (float) ((scaledResolution.getScaledHeight()) - (ClientFont.INSTANCE.isEnabled() ? 12 : 11)), -1);
        }
    }

    public void mouseClicked(int mouseX, int mouseY) {
        icons.forEach(icon -> icon.whenClicked(mouseX, mouseY));
    }

    public String getTooltip() {
        return tooltip;
    }

    public void setTooltip(String tooltipIn) {
        if (Objects.equals(tooltipIn, "")) {
            tooltipAnimation.setState(false);
        }

        else {
            tooltip = tooltipIn;
            tooltipAnimation.setState(true);
        }
    }

}

File: src/main/java/com/paragon/client/ui/taskbar/Taskbar.kt
package com.paragon.client.ui.taskbar

import com.paragon.api.util.Wrapper
import com.paragon.api.util.render.RenderUtil
import com.paragon.client.systems.module.impl.client.Colours
import com.paragon.client.ui.taskbar.start.StartMenu
import com.paragon.client.ui.util.Click
import com.paragon.client.ui.util.animation.Animation
import com.paragon.client.ui.util.animation.Easing
import net.minecraft.client.gui.ScaledResolution
import java.awt.Color

/**
 * @author Surge
 * @since 26/07/2022
 */
object Taskbar : Wrapper {

    var tooltip = ""

    val expandAnimation: Animation = Animation({ 500f }, false, { Easing.CUBIC_IN_OUT })
    val timer: Animation = Animation( { 1000f }, false, { Easing.LINEAR })

    fun draw(mouseX: Int, mouseY: Int) {
        val scaledResolution = ScaledResolution(minecraft)
        val hovered = mouseY.toFloat() in scaledResolution.scaledHeight - 80f..scaledResolution.scaledHeight.toFloat()

        var state = false

        if (StartMenu.expandAnimation.state || hovered) {
            state = true
        }

        expandAnimation.state = state

        RenderUtil.drawRect(0f, scaledResolution.scaledHeight - (26f * expandAnimation.getAnimationFactor().toFloat()), scaledResolution.scaledWidth.toFloat(), 26f, Color(148, 148, 148).rgb)

        StartMenu.x = 2f
        StartMenu.y = scaledResolution.scaledHeight - (23f * expandAnimation.getAnimationFactor().toFloat())
        StartMenu.draw(mouseX, mouseY)

        RenderUtil.drawRect(150f, scaledResolution.scaledHeight - (26f * expandAnimation.getAnimationFactor().toFloat()) - 2, scaledResolution.scaledWidth.toFloat() - 150f, 2f, Colours.mainColour.value.rgb)
    }

    fun mouseClicked(mouseX: Int, mouseY: Int, click: Click) {
        if (!StartMenu.mouseClicked(mouseX, mouseY, click)) {
            StartMenu.expandAnimation.state = false
        }
    }

}